### PR TITLE
upgraded to latest node lib - csrf feature

### PR DESCRIPTION
**Pull request by paddybasi**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-1957

### Change description ###
upgrade node lib to implement csrf feature

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: api/auth/index.ts [hunk 1]
```diff
@@ -67,7 +67,7 @@ export const getXuiNodeMiddleware = () => {
         callbackURL: getConfigValue(SERVICES_IDAM_OAUTH_CALLBACK_URL),
         clientID: idamClient,
         clientSecret: secret,
-        discoveryEndpoint: `${idamWebUrl}/o`,
+        discoveryEndpoint: `${idamWebUrl}/o/.well-known/openid-configuration`,
```
Comment (ronaldmansveld) [inline api/auth/index.ts:70]: directories that start with a . are generally hidden/system dirs. Are we sure we are supposed to hard code it this way?

**Review by Mo-Lala84 — APPROVED** [review]
lgtm

File: api/auth/index.ts (hunk 1 above)
Comment (paddybasi) [reply api/auth/index.ts:70]: it's not a directory, but the standard for autodiscovery endpoints @see https://ldapwiki.com/wiki/Openid-configuration#:~:text=Openid%2Dconfiguration%20is%20a%20Well,the%20Identity%20Provider%20(IDP).

Comment (paddybasi) [reply api/auth/index.ts:70]: However I do agree that perhaps it shouldn't be hardcoded, maybe a task in future to move this into the config instead

Comment (Mo-Lala84) [reply api/auth/index.ts:70]: its inline with guidelines, not a show stopper but a good call to query it.

**Review by ronaldmansveld — APPROVED** [review]

**Pull request merged**